frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -797,7 +816,13 @@ pub mod pallet {
                 ));
             }
 
-            T::Currency::resolve_creating(&staker, reward_imbalance);
+            // Check if the staker has a reward destination account or not
+            // let mut reward_received_account = staker.clone();
+
+            let reward_received_account =
+                Self::check_next_delegate_reward(&staker, &contract_id, 0);
+            T::Currency::resolve_creating(&reward_received_account, reward_imbalance);
```

Comment:
What if reward destination is set to `StakeBalance`?

Reply:
Now it should be working ok .. and if the reward destination is set to StakeBalance it would re-stake the rewards received.